packages/java-edition/src/mcfunction/checker/index.ts

```diff
@@ -162,14 +166,46 @@ const entity: core.SyncChecker<EntityNode> = (node, ctx) => {
 }
 
 const item: core.SyncChecker<ItemNode> = (node, ctx) => {
-	if (!node.nbt) {
+	if (!ItemNode.hasUserData(node)) {
 		return
 	}
 
-	nbt.checker.index(
-		'item',
-		core.ResourceLocationNode.toString(node.id, 'full'),
-	)(node.nbt, ctx)
+	if (ItemOldNode.is(node)) {
+		nbt.checker.index(
+			'item',
+			core.ResourceLocationNode.toString(node.id, 'full'),
+		)(node.nbt!, ctx)
+	} else {
+		const groupedComponents = new Map<
+			string,
+			core.PairNode<core.ResourceLocationNode, nbt.NbtNode>[]
+		>()
+
+		node.components!.children.forEach(component => {
+			const componentName = core.ResourceLocationNode.toString(
+				component.key!,
+				'full',
+			)
+
+			if (!groupedComponents.has(componentName)) {
+				groupedComponents.set(componentName, [])
+			}
+
+			groupedComponents.get(componentName)!.push(component)
+		})
+
+		groupedComponents.forEach((components, componentName) => {
+			if (components.length > 1) {
+				components.forEach(component => {
+					ctx.err.report(
+						localize('duplicate-components', componentName),
+						component.key!.range,
+						core.ErrorSeverity.Error,
+					)
+				})
+			}
```

Comment:
You can use `core.LanguageErrorInfo.related` to report only one error for this with the duplicated keys' locations added in the `info.related` field (`core.Location.create()` would be helpful to create the entries for the field)

Reply:
Vscode also creates an error for each duplicate key in a json file for example, imo this is the most clean